src/debugger/Debugger.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { BabelNode } from "babel-types";
+import { Breakpoint } from "./Breakpoint.js";
+import invariant from "../invariant.js";
+import { DebugChannel } from "../DebugChannel.js";
+
+export class Debugger {
+  constructor(channel: DebugChannel) {
+    this.breakpoints = new Map();
+    this.prevBreakLine = 0;
+    this.prevBreakCol = 0;
+    this.channel = channel;
+    this.waitForRun(1000, function(line) {
+      return line === "Run";
+    });
+  }
+  breakpoints: { [number]: Breakpoint };
```

Comment:
Add some comments to each of the properties to document intent. For example, the index is not just a number, it is a line number. Which immediately raises the question: what about the file and the column?